dev/NumberBox/NumberBoxParser.h

```diff
@@ -0,0 +1,78 @@
+﻿#pragma once
+#include <vector>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <cmath>
+#include <algorithm>
+#include <iterator>
+#include <cstdlib>
+#include <regex>
+#include <stack>
+#include <optional>
+
+// Represents a token in a mathematic expression. This can be a simple operator (+-*/^), a decimal number, or parentheses. 
+class MathToken
+{
+public:
+    enum TokenType
+    {
+        Invalid = 0,
+        Numeric = 1,
+        Operator = 2,
+        ParenLeft = 3,
+        ParenRight = 4,
+        EOFToken = 6
+    };
+    TokenType type;
+    std::wstring str;
+    MathToken();
```

Comment:
Looks like this type [shouldn't have a default constructor](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c45-dont-define-a-default-constructor-that-only-initializes-data-members-use-in-class-member-initializers-instead) but if you want one then you can just use the default with MathToken() = default;